<h5>Esta análise tem como finalidade mostrar gráficos com rendas e QI de 70 países</h5>
<h5>Fontes: (consultada em 08/06/2023)</h5>
<ul>
<li>https://www.dadosmundiais.com/europa/malta/index.php</li>
<li>https://www.dadosmundiais.com/renda-media.php</li>
</ul>


In [1]:
# importando as bibliotecas que serão usadas
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

In [2]:
# carregando a base de dados e visualizando os 5 primeiros registros
base_qi = pd.read_excel('qi_paises.xlsx')
base_qi.head()

INDICE        PAIS  RENDA_ANUAL(USD)   QI
0       1       Suíça             90600  100
1       2  Luxemburgo             88190  100
2       3     Noruega             83880   98
3       4     Irlanda             76110   95
4       5         USA             70930   97

In [3]:
# Excluindo a coluna INDICE, que não terá utilidade
base_qi = base_qi.drop(['INDICE'], axis=1)
base_qi.head()

PAIS  RENDA_ANUAL(USD)   QI
0       Suíça             90600  100
1  Luxemburgo             88190  100
2     Noruega             83880   98
3     Irlanda             76110   95
4         USA             70930   97

In [4]:
# Informações sobre a base de dados
base_qi.info

<bound method DataFrame.info of                  PAIS  RENDA_ANUAL(USD)   QI
0               Suíça             90600  100
1          Luxemburgo             88190  100
2             Noruega             83880   98
3             Irlanda             76110   95
4                 USA             70930   97
..                ...               ...  ...
65  Congo (dem. rep.)               550   64
66         Moçambique               480   71
67        Afeganistão               390   81
68              Malta             30760   91
69            Croácia             17630   95

[70 rows x 3 columns]>

In [5]:
# Estatísticas
base_qi.describe()

RENDA_ANUAL(USD)          QI
count         70.000000   70.000000
mean       25978.285714   88.242857
std        25709.529557   12.292847
min          390.000000   56.000000
25%         4135.000000   81.000000
50%        14470.000000   91.000000
75%        46145.000000   98.000000
max        90600.000000  106.000000

In [6]:
# Relação entre Renda e QI
# Gráfico interativo com plotly
fig = px.scatter(base_qi, x='RENDA_ANUAL(USD)', y='QI', trendline='ols')
fig.show()

Pelo gráfico acima a um primeiro olhar podemos afirmar que o qi aumenta conforme maior a renda anual do país, mas se observarmos bem, o aumento da renda anual não necessariamente implica no maior qi médio da população haja visto que temos um ponto no gráfico com Qi = 106 e com renda anual = 64 K e temos um ponto com QI = 106 e renda anual = 7.090 K. 

In [7]:
# 5 Países com maiores QIs
paises_maiores_qis = base_qi['QI'].sort_values(ascending=False).head(5)
paises_maiores_qis

42    106
21    106
6     106
11    106
35    104
Name: QI, dtype: int64

In [27]:
# Impriminodo os 5 países com os maiores QIs
print(base_qi.iloc[42]['PAIS'], '\n',base_qi.iloc[21]['PAIS'],'\n',base_qi.iloc[6]['PAIS'],'\n',base_qi.iloc[11]['PAIS'],'\n',base_qi.iloc[35]['PAIS'])

Tailândia 
 Japão 
 Cingapura 
 Hong Kong  
 China


In [39]:
# E o Brasil?
# Qual a renda anual e o Qi médio?
print(base_qi.loc[base_qi['PAIS'] == 'Brasil'])

      PAIS  RENDA_ANUAL(USD)  QI
41  Brasil              7740  83


In [41]:
# Criando um mapa interativo entre renda, qi e País
grafico = px.treemap(base_qi, path=['RENDA_ANUAL(USD)', 'QI', 'PAIS'])
grafico.show()